server/runtime_javascript_nakama.go

```diff
@@ -8004,7 +8004,15 @@ func (n *runtimeJavascriptNakamaModule) localcachePut(r *goja.Runtime) func(goja
 			panic(r.NewTypeError("ttl must be 0 or more"))
 		}
 
-		n.localCache.Put(key, value.Export(), ttl)
+		v := value.Export()
+
+		switch v.(type) {
+		case string, int64, float64, bool:
+		default:
+			panic(r.NewTypeError("unsupported value type: must be string, numeric or boolean"))
```

Comment:
Is there a chance this would cause panics in existing modules that use types outside of this list?
I.e. would it make sense to issue a warning/error instead of panicking to prevent breaking backward compatibility?

Reply:
Goja types that are passed by reference (e.g.: a JS object) carry with them internally a reference to the original VM where they were instantiated, if these objects are manipulated concurrently it can cause the VM to panic.